class: Workflow
cwlVersion: v1.0
id: qc_collapsed_bam
label: qc_collapsed_bam
$namespaces:
  sbg: 'https://www.sevenbridges.com/'
inputs:
  - id: reference
    type: File
    secondaryFiles:
      - ^.fasta.fai
      - ^.dict
    'sbg:x': -1379.3106689453125
    'sbg:y': -9
  - id: pool_b_target_intervals
    type: File
    label: pool_b_target_intervals
    'sbg:x': -1380.0771484375
    'sbg:y': -176.342529296875
  - id: pool_a_targets_intervals
    type: File
    label: pool_a_targets_intervals
    'sbg:x': -1407.2725830078125
    'sbg:y': -725.6703491210938
  - id: vcf_file
    type: File
    doc: VCF file containing the SNPs to be queried.
    'sbg:x': -1376.3106689453125
    'sbg:y': 160.8839111328125
  - id: collapsed_bam
    type:
      - File
      - type: array
        items: File
    label: collapsed_bam
    'sbg:x': -899.7366333007812
    'sbg:y': 462.836181640625
  - id: sample_type
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: 'Sample types: Normal or Tumor.'
    'sbg:x': -900.1541137695312
    'sbg:y': 613.905517578125
  - id: sample_sex
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: Expected sample sex (i.e. M or F).
    'sbg:x': -909.779296875
    'sbg:y': 779.0851440429688
  - id: sample_name
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: >-
      Sample name. If not specified, sample name is automatically figured out
      from the BAM file.
    'sbg:x': -913.1387939453125
    'sbg:y': 910.29150390625
  - id: sample_group
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: The sample group (e.g. the sample patient ID).
    'sbg:x': -907.56005859375
    'sbg:y': 1060.3988037109375
  - id: group_reads_by_umi_bam
    type:
      - File
      - type: array
        items: File
    label: group_reads_by_umi_bam
    doc: Input BAM file generated by GroupReadByUmi.
    'sbg:x': -911.3615112304688
    'sbg:y': 324.525634765625
  - id: pool_a_baits_intervals
    type: File?
    label: pool_a_baits_intervals
    doc: 'Optional set of intervals over which to restrict analysis. [Optional].'
    'sbg:x': -1419.413818359375
    'sbg:y': -885.5172119140625
  - id: pool_b_baits_intervals
    type: File?
    label: pool_b_baits_intervals
    doc: 'Optional set of intervals over which to restrict analysis. [Optional].'
    'sbg:x': -1393.6268310546875
    'sbg:y': -352.0533142089844
  - id: bed_file
    type: File?
    doc: BED file containing the intervals to be queried.
    'sbg:x': -1384.4722900390625
    'sbg:y': 347.15325927734375
outputs:
  - id: biometrics_extract_pickle
    outputSource:
      - biometrics_extract/biometrics_extract_pickle
    type:
      - File
      - type: array
        items: File
    'sbg:x': 1073.549560546875
    'sbg:y': -58.49618148803711
  - id: fgbio_collect_duplex_seq_metrics_duplex_family_size
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_family_size
    type: File
    'sbg:x': 516.4937744140625
    'sbg:y': -1038.1038818359375
  - id: fgbio_collect_duplex_seq_metrics_duplex_qc
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_qc
    type: File?
    'sbg:x': 507.3194885253906
    'sbg:y': -1158.8990478515625
  - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    type: File?
    'sbg:x': 510.3775939941406
    'sbg:y': -1284.28125
  - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
    type: File
    'sbg:x': 514.9647216796875
    'sbg:y': -1418.837890625
  - id: fgbio_collect_duplex_seq_metrics_family_size
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_family_size
    type: File
    'sbg:x': 516.4937744140625
    'sbg:y': -1547.2781982421875
  - id: fgbio_collect_duplex_seq_metrics_umi_counts
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_umi_counts
    type: File
    'sbg:x': 516.4937744140625
    'sbg:y': -1683.3638916015625
  - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_family_size
    type: File
    'sbg:x': 510.3775939941406
    'sbg:y': -1842.38525390625
  - id: fgbio_collect_duplex_seq_metrics_duplex_qc_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_qc
    type: File?
    'sbg:x': 508.8485412597656
    'sbg:y': -1998.3485107421875
  - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    type: File?
    'sbg:x': 499.6742248535156
    'sbg:y': -2158.89892578125
  - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
    type: File
    'sbg:x': 495.0870666503906
    'sbg:y': -2319.449462890625
  - id: fgbio_collect_duplex_seq_metrics_family_size_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_family_size
    type: File
    'sbg:x': 502.7323303222656
    'sbg:y': -2457.064208984375
  - id: fgbio_collect_duplex_seq_metrics_umi_counts_1
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_umi_counts
    type: File
    'sbg:x': 498.1451721191406
    'sbg:y': -2605.382080078125
  - id: biometrics_minor_csv
    outputSource:
      - biometrics_minor/biometrics_minor_csv
    type: File
    'sbg:x': 1108.409912109375
    'sbg:y': 879.7926025390625
  - id: biometrics_minor_json
    outputSource:
      - biometrics_minor/biometrics_minor_json
    type: File?
    'sbg:x': 1099.409912109375
    'sbg:y': 734.1456909179688
  - id: biometrics_minor_plot
    outputSource:
      - biometrics_minor/biometrics_minor_plot
    type: File?
    'sbg:x': 1092.851806640625
    'sbg:y': 577.2938232421875
  - id: biometrics_minor_sites_plot
    outputSource:
      - biometrics_minor/biometrics_minor_sites_plot
    type: File?
    'sbg:x': 1085.1456298828125
    'sbg:y': 460.587646484375
  - id: biometrics_major_csv
    outputSource:
      - biometrics_major/biometrics_major_csv
    type: File
    'sbg:x': 1085.49560546875
    'sbg:y': 332.7539367675781
  - id: biometrics_major_json
    outputSource:
      - biometrics_major/biometrics_major_json
    type: File?
    'sbg:x': 1089.466064453125
    'sbg:y': 211.00634765625
  - id: biometrics_major_plot
    outputSource:
      - biometrics_major/biometrics_major_plot
    type: File?
    'sbg:x': 1077.554931640625
    'sbg:y': 82.63099670410156
  - id: biometrics_sexmismatch_json
    outputSource:
      - biometrics_sexmismatch/biometrics_sexmismatch_json
    type: File?
    'sbg:x': 1105.6673583984375
    'sbg:y': 1000.427490234375
  - id: biometrics_sexmismatch_csv
    outputSource:
      - biometrics_sexmismatch/biometrics_sexmismatch_csv
    type: File
    'sbg:x': 1098.1990966796875
    'sbg:y': 1146.0594482421875
  - id: gatk_collect_insert_size_metrics_txt
    outputSource:
      - bam_qc_stats_1/gatk_collect_insert_size_metrics_txt
    type: File
    'sbg:x': 248.5195770263672
    'sbg:y': -892.0381469726562
  - id: gatk_collect_insert_size_metrics_histogram_pdf
    outputSource:
      - bam_qc_stats_1/gatk_collect_insert_size_metrics_histogram_pdf
    type: File
    'sbg:x': 251.14996337890625
    'sbg:y': -774.7000732421875
  - id: gatk_collect_hs_metrics_txt
    outputSource:
      - bam_qc_stats_1/gatk_collect_hs_metrics_txt
    type: File
    'sbg:x': 254.2693634033203
    'sbg:y': -660.8965454101562
  - id: gatk_collect_hs_metrics_per_target_coverage_txt
    outputSource:
      - bam_qc_stats_1/gatk_collect_hs_metrics_per_target_coverage_txt
    type: File
    'sbg:x': 252.1963348388672
    'sbg:y': -549.349609375
  - id: gatk_collect_hs_metrics_per_base_coverage_txt
    outputSource:
      - bam_qc_stats_1/gatk_collect_hs_metrics_per_base_coverage_txt
    type: File
    'sbg:x': 255.41932678222656
    'sbg:y': -435.50469970703125
  - id: gatk_collect_alignment_summary_metrics_txt
    outputSource:
      - bam_qc_stats_1/gatk_collect_alignment_summary_metrics_txt
    type: File
    'sbg:x': 258.86920166015625
    'sbg:y': -320.5088806152344
  - id: gatk_collect_insert_size_metrics_txt_1
    outputSource:
      - bam_qc_stats/gatk_collect_insert_size_metrics_txt
    type: File
    'sbg:x': 676.3040771484375
    'sbg:y': -586.1397094726562
  - id: gatk_collect_insert_size_metrics_histogram_pdf_1
    outputSource:
      - bam_qc_stats/gatk_collect_insert_size_metrics_histogram_pdf
    type: File
    'sbg:x': 684.9093017578125
    'sbg:y': -452.3918151855469
  - id: gatk_collect_hs_metrics_txt_1
    outputSource:
      - bam_qc_stats/gatk_collect_hs_metrics_txt
    type: File
    'sbg:x': 680.3305053710938
    'sbg:y': -319.60614013671875
  - id: gatk_collect_hs_metrics_per_target_coverage_txt_1
    outputSource:
      - bam_qc_stats/gatk_collect_hs_metrics_per_target_coverage_txt
    type: File
    'sbg:x': 680.3305053710938
    'sbg:y': -182.2416534423828
  - id: gatk_collect_hs_metrics_per_base_coverage_txt_1
    outputSource:
      - bam_qc_stats/gatk_collect_hs_metrics_per_base_coverage_txt
    type: File
    'sbg:x': 678.0410766601562
    'sbg:y': -59.758304595947266
  - id: gatk_collect_alignment_summary_metrics_txt_1
    outputSource:
      - bam_qc_stats/gatk_collect_alignment_summary_metrics_txt
    type: File
    'sbg:x': 683.7645874023438
    'sbg:y': 53.56740951538086
  - id: biometrics_genotype_plot_input_database
    outputSource:
      - biometrics_genotype/biometrics_genotype_plot_input_database
    type: File?
    'sbg:x': 1109.0440673828125
    'sbg:y': 1322.331787109375
  - id: biometrics_genotype_plot_input
    outputSource:
      - biometrics_genotype/biometrics_genotype_plot_input
    type: File?
    'sbg:x': 1110.3228759765625
    'sbg:y': 1477.0633544921875
  - id: biometrics_genotype_comparisons
    outputSource:
      - biometrics_genotype/biometrics_genotype_comparisons
    type: File
    'sbg:x': 1119.2742919921875
    'sbg:y': 1657.3702392578125
  - id: biometrics_genotype_cluster_input_database
    outputSource:
      - biometrics_genotype/biometrics_genotype_cluster_input_database
    type: File?
    'sbg:x': 1128.2257080078125
    'sbg:y': 1828.7257080078125
  - id: biometrics_genotype_cluster_input
    outputSource:
      - biometrics_genotype/biometrics_genotype_cluster_input
    type: File
    'sbg:x': 1132.06201171875
    'sbg:y': 2006.47509765625
steps:
  - id: bam_qc_stats
    in:
      - id: input
        source: collapsed_bam
      - id: target_intervals
        source: pool_b_target_intervals
      - id: bait_intervals
        source: pool_b_baits_intervals
      - id: reference
        source: reference
    out:
      - id: gatk_collect_insert_size_metrics_histogram_pdf
      - id: gatk_collect_insert_size_metrics_txt
      - id: gatk_collect_hs_metrics_txt
      - id: gatk_collect_hs_metrics_per_base_coverage_txt
      - id: gatk_collect_hs_metrics_per_target_coverage_txt
      - id: gatk_collect_alignment_summary_metrics_txt
    run: ../bam_qc_stats/bam_qc_stats.cwl
    label: bam_qc_stats
    'sbg:x': 244.58816528320312
    'sbg:y': -98.25187683105469
  - id: bam_qc_stats_1
    in:
      - id: input
        source: collapsed_bam
      - id: target_intervals
        source: pool_a_targets_intervals
      - id: bait_intervals
        source: pool_a_baits_intervals
      - id: reference
        source: reference
    out:
      - id: gatk_collect_insert_size_metrics_histogram_pdf
      - id: gatk_collect_insert_size_metrics_txt
      - id: gatk_collect_hs_metrics_txt
      - id: gatk_collect_hs_metrics_per_base_coverage_txt
      - id: gatk_collect_hs_metrics_per_target_coverage_txt
      - id: gatk_collect_alignment_summary_metrics_txt
    run: ../bam_qc_stats/bam_qc_stats.cwl
    label: bam_qc_stats
    'sbg:x': -79.8152847290039
    'sbg:y': -635.7706909179688
  - id: biometrics_extract
    in:
      - id: sample_bam
        source:
          - collapsed_bam
      - id: sample_type
        source:
          - sample_type
      - id: sample_sex
        source:
          - sample_sex
      - id: sample_group
        source:
          - sample_group
      - id: sample_name
        source:
          - sample_name
      - id: fafile
        source: reference
      - id: vcf_file
        source: vcf_file
      - id: bed_file
        source: bed_file
    out:
      - id: biometrics_extract_pickle
    run: ../command_line_tools/biometrics_extract/0.2.7/biometrics_extract.cwl
    'sbg:x': -56.18357467651367
    'sbg:y': 437.74395751953125
  - id: fgbio_collect_duplex_seq_metrics_1_2_0
    in:
      - id: input
        source: group_reads_by_umi_bam
      - id: intervals
        source: pool_a_baits_intervals
    out:
      - id: fgbio_collect_duplex_seq_metrics_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
      - id: fgbio_collect_duplex_seq_metrics_umi_counts
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    run: >-
      ../command_line_tools/fgbio_collect_duplex_seq_metrics_1.2.0/fgbio_collect_duplex_seq_metrics_1.2.0.cwl
    label: fgbio_collect_duplex_seq_metrics_1.2.0
    'sbg:x': -102.85713958740234
    'sbg:y': -1250.857177734375
  - id: fgbio_collect_duplex_seq_metrics_1_2_1
    in:
      - id: input
        source: group_reads_by_umi_bam
      - id: intervals
        source: pool_b_baits_intervals
    out:
      - id: fgbio_collect_duplex_seq_metrics_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
      - id: fgbio_collect_duplex_seq_metrics_umi_counts
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    run: >-
      ../command_line_tools/fgbio_collect_duplex_seq_metrics_1.2.0/fgbio_collect_duplex_seq_metrics_1.2.0.cwl
    label: fgbio_collect_duplex_seq_metrics_1.2.0
    'sbg:x': -101.767578125
    'sbg:y': -2137.599365234375
  - id: biometrics_major
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
    out:
      - id: biometrics_major_csv
      - id: biometrics_major_json
      - id: biometrics_major_plot
    run: ../command_line_tools/biometrics_major/0.2.7/biometrics_major.cwl
    'sbg:x': 677.335205078125
    'sbg:y': 262.2733154296875
  - id: biometrics_minor
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
      - id: plot
        default: false
      - id: json
        default: true
    out:
      - id: biometrics_minor_csv
      - id: biometrics_minor_json
      - id: biometrics_minor_plot
      - id: biometrics_minor_sites_plot
    run: ../command_line_tools/biometrics_minor/0.2.7/biometrics_minor.cwl
    'sbg:x': 686.5601196289062
    'sbg:y': 571.31689453125
  - id: biometrics_sexmismatch
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
    out:
      - id: biometrics_sexmismatch_csv
      - id: biometrics_sexmismatch_json
    run: >-
      ../command_line_tools/biometrics_sexmismatch/0.2.7/biometrics_sexmismatch.cwl
    'sbg:x': 688.3497924804688
    'sbg:y': 1029.9459228515625
  - id: biometrics_genotype
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
    out:
      - id: biometrics_genotype_comparisons
      - id: biometrics_genotype_cluster_input
      - id: biometrics_genotype_cluster_input_database
      - id: biometrics_genotype_plot_input
      - id: biometrics_genotype_plot_input_database
    run: ../command_line_tools/biometrics_genotype/0.2.7/biometrics_genotype.cwl
    'sbg:x': 730.717529296875
    'sbg:y': 1501
requirements:
  - class: SubworkflowFeatureRequirement
  - class: InlineJavascriptRequirement
  - class: StepInputExpressionRequirement
